src/external-server/docker-amarna.ts

```diff
@@ -0,0 +1,42 @@
+import { Image } from "@nomiclabs/hardhat-docker";
+import { DockerServer } from "./docker-server";
+import * as fs from "fs";
+
+export class AmarnaDocker extends DockerServer {
+    useShell = false;
+    /**
+     * @param image the Docker image to be used for running the container
+     * @param cairoPaths the paths specified in hardhat config cairoPaths
+     */
+    constructor(image: Image, private path: string) {
+        super(image, "127.0.0.1", null, "", "starknet-docker-proxy");
+    }
+
+    protected async getDockerArgs(): Promise<string[]> {
+        // To access the files on host machine from inside the container, proper mounting has to be done.
+        const volumes = ["-v", `${this.path}:/src`];
+        const dockerArgs = [...volumes];
+        if (this.useShell) {
+            // Run ./amarna.sh file for custom args
+            if (fs.existsSync(`${this.path}/amarna.sh`)) {
```

Comment:
Is this some amarna native feature? If so, where is it documented? If not, what is it?

Reply:
It's in case some projs wants to use some of the ton of config options Amarna provides.

Reply:
Ok, sounds useful, let's keep it as it is, but documented!

Reply:
Aye cap!